src/coreclr/jit/importer.cpp

```diff
@@ -4503,13 +4503,123 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
+                        {
+                            retNode = gtNewSimdHWIntrinsicNode(callType, op2, NI_Vector128_ToScalar, callJitType,
+                                                               insSimdSize);
+                            break;
+                        }
+                    }
+
+                    if (ni == NI_System_Math_Min)
+                    {
+                        // 3. if both inputs are zero
+                        if (op1->IsFloatPositiveZero() && op2->IsFloatPositiveZero())
+                        {
+                            // IsFloatPositiveZero doesn't gurantee that we don't have op1 = -0.0 op2 = -0.0,
+                            // so we check the sign.
+                            // Ensuring -0.0 is returned if both inputs are zero since both inputs being zero,
+                            // of either sign, means the second argument is returned
+                            //
+                            // 3.1 if op1 is -0.0 and op2 is either sign then take op1
+                            if (op1->IsFloatNegativeZero())
+                            {
+                                retNode = gtNewSimdHWIntrinsicNode(callType, op1, NI_Vector128_ToScalar, callJitType,
+                                                                   insSimdSize);
+                                break;
+                            }
+                            // 3.2 if op2 is -0.0 and op1 is either sign then take op2
+                            else if (op2->IsFloatNegativeZero())
+                            {
+                                retNode = gtNewSimdHWIntrinsicNode(callType, op2, NI_Vector128_ToScalar, callJitType,
+                                                                   insSimdSize);
+                                break;
+                            }
+                        }
+
+                        //  3.3. if op1 is -0.0 then take op2
+                        //  3.4. if op2 is -0.0 then take op2
+                        if (op1->IsFloatNegativeZero() || op2->IsFloatNegativeZero())
+                        {
+                            retNode = gtNewSimdHWIntrinsicNode(callType, op2, NI_Vector128_ToScalar, callJitType,
+                                                               insSimdSize);
+                            break;
+                        }
+                    }
+
+                    impPushOnStack(op1, callType);
+                    impPushOnStack(op2, callType);
+                    break;
+                    //// If it's any other constant then it needs to be op1 for both (Math.Min/Math.Max).
+                    // retNode = gtNewSimdHWIntrinsicNode(callType, op1, NI_Vector128_ToScalar, callJitType,
+                    // insSimdSize);
+                    // break;
+                }
+
+                // Go down to the direct Math.Min/Math.Max call
+                // break;
+            }
```

Comment:
@tannergooding @jakobbotsch 
Please take a look at the implementation for +0.0, -0.0 and nan. Do I understand it right?
I got some questions about next situations:
What if we got op1=-0.0 op2=-0.0 and max? Should it go to the Math.Max call path?
What if we got op1=+0.0 op2 =+0.0 and min? Should it go to the Math.Min call path?
I got confused about the next: "If its any other constant then it needs to be op1 for both." which basically means that if we got const=op1=-100 and variable=op2=200 and max, we should return op1=-100. It doesn't seem right, does it?

Reply:
P.S. It is not completed yet, I assumed there would be failures. I want to make sure I got the +0.0 -0.0 nan implementation right.

Reply:
When both are constant we can actually constant fold it directly and avoid creating an intrinsic node at all.

The comments are for when only one is constant.

Reply:
So:
1. if both are constant and none fits the conditions, then I push the ops back on the stack and let it go. Right?

2. I'm still confused about the quote. What if we have:

double x = 10.0;

public double Test(double x) => Math.Max(x, 15.0);

double result = Test(x) 

Where:
15.0 is op2 and constant.
10.0 is op1 and unknown.
Then according to the quote we have to return 10, but it doesn't seem right.

Reply:
`x` is unknown and `y` is known to be 15.

This means we'll emit `maxsd 15, x`. If `x` is `NaN` then it will be returned because its `op2`. `If `x` is any other value, then `maxsd 15, x` will return whichever is larger.